charts/datadog/templates/_container-system-probe.yaml

```diff
@@ -65,7 +65,7 @@
       mountPropagation: {{ .Values.datadog.hostVolumeMountPropagation }}
       readOnly: true
     {{- include "linux-container-host-release-volumemounts" . | nindent 4 }}
-  {{- if .Values.datadog.systemProbe.enableDefaultOsReleasePaths }}
+    {{- if or .Values.datadog.systemProbe.enableDefaultOsReleasePaths (not .Values.datadog.disableDefaultOsReleasePaths) }}
```

Comment:
I think this might need to be `and`, otherwise the default value of `true` for the deprecated `datadog.systemProbe.enableDefaultOsReleasePaths` will always ignore the new config option.